src/pipeline/rackhd/ova/build.sh

```diff
@@ -0,0 +1,137 @@
+#!/bin/bash -x
+
+set +e
+
+buildOva(){
+  # If OVA_CACHE_BUILD is not used, cache_image directory does not exist
+  if [ -d  $WORKSPACE/$CACHE_IMAGE_DIR/RackHD/packer/ ] ; then
+      echo "Copy Cache images from PACKER_CACHE_BUILD job archiving"
+      mv $WORKSPACE/$CACHE_IMAGE_DIR/RackHD/packer/* $RACKHD_DIR/packer/
+      ls $RACKHD_DIR/packer/*
+  fi
+
+  vmware -v
+
+  echo using artifactory : $ARTIFACTORY_URL
+
+  source ./$ON_BUILD_CONFIG_DIR/shareMethod.sh $ON_BUILD_CONFIG_DIR
+
+  pushd $RACKHD_DIR/packer/ansible/roles/rackhd-builds/tasks
+  sed -i "s#https://dl.bintray.com/rackhd/debian trusty release#${ARTIFACTORY_URL}/${STAGE_REPO_NAME} ${DEB_DISTRIBUTION} ${DEB_COMPONENT}#" main.yml
+  sed -i "s#https://dl.bintray.com/rackhd/debian trusty main#${ARTIFACTORY_URL}/${STAGE_REPO_NAME} ${DEB_DISTRIBUTION} ${DEB_COMPONENT}#" main.yml
+  popd
+
+  echo "kill previous running packer instances"
+
+  set +e
+  pkill packer
+  pkill vmware
+  set -e
+
+  pushd $RACKHD_DIR/packer
+  echo "Start to packer build .."
+
+  export PACKER_CACHE_DIR=$HOME/.packer_cache
+  #export vars to build ova
+  if [ "${IS_OFFICIAL_RELEASE}" == true ]; then
+      export ANSIBLE_PLAYBOOK=rackhd_release
+  else
+      export ANSIBLE_PLAYBOOK=rackhd_ci_builds
+  fi
+
+  if [ "$BUILD_TYPE" == "vmware" ] &&  [ -f output-vmware-iso/*.vmx ]; then
+       echo "Build from template cache"
+       export BUILD_STAGE=BUILD_FINAL
+  else
+       echo "Build from begining"
+       export BUILD_STAGE=BUILD_ALL
+  fi
+
+  export RACKHD_VERSION=$RACKHD_VERSION
+  #export end
+
+  ./HWIMO-BUILD
```

Comment:
The script mixed up so much function: build vagrant box, build vagrant ovf, build ova, build vmx.
It's hard to read.
I think it's better to implement a independent shell script build_ova.sh
And it only need parameters: os_ver, rackhd_version, rackhd_apt_repository, [cache_image]